src/kiota/Rpc/SecurityRequirementMapper.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Microsoft.OpenApi.Models;
+
+namespace kiota.Rpc
+{
+    internal class SecurityRequirementMapper
+    {
+        public static IDictionary<string, SecurityRequirement>? FromSecurityRequirementList(IList<OpenApiSecurityRequirement>? securityRequirementList)
+        {
+            var requirements = new Dictionary<string, SecurityRequirement>();
+            if (securityRequirementList is null) return null;
+
+            foreach (var securityRequirement in securityRequirementList)
+            {
+                if (securityRequirement is null) continue;
+
+                foreach (var securityRequirementItem in securityRequirement)
+                {
+                    if (securityRequirementItem.Key.Reference is null) continue;
+                    if (securityRequirementItem.Value is null) continue;
+                    if (securityRequirementItem.Key.Reference.Id is null) continue;
+
+                    string name = securityRequirementItem.Key.Reference.Id;
+                    var scopes = new List<string>();
+                    foreach (var scope in securityRequirementItem.Value)
+                    {
+                        scopes.Add(scope);
+                    }
```

Comment:
```suggestion
                    var scopes = securityRequirementItem.Value.ToList()
```

Could we leverage Linq's ToList here to avoid the loop?

Reply:
Addressed at [44a21a5](https://github.com/microsoft/kiota/pull/6172/commits/44a21a51522f5a7d9c21d9091971d598449ba2d7)

Reply:
Having a mapper function helps here for maintenance and readability of the code